src/unix/core.c

```diff
@@ -1284,6 +1288,73 @@ int uv_translate_sys_error(int sys_errno) {
 }
 
 
+int uv_os_environ(uv_env_item_t** envitems, int* count) {
+  int i, j, cnt;
+  uv_env_item_t* envitem;
+
+  *envitems = NULL;
+  *count = 0;
+
+  for (i = 0; environ[i]; i++);
+
+  *envitems = uv__calloc(i, sizeof(**envitems));
```

Comment:
Don't you want to allocate an extra slot to store a trailing NULL pointer? I think that's what most people will expect since it otherwise mirrors `environ` quite closely.

Users then don't have to pass in the count to `uv_os_free_environ()` either.

Reply:
I started there, but then noticed uv_cpu_info and uv_interface_addresses and thought "maybe let's make it consistent?" But I think you're right and making it mimic `environ` makes more sense here. That means I can skip the count parameter on both.

Reply:
WDYT?

Reply:
I couldn't figure out a nice way to do this that wouldn't complicate things so I left the count argument and thus didn't add an extra space.